File: src/atcoder/dp/ProblemC.java
package atcoder.dp;

import java.util.Scanner;

public class ProblemC {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] prev = new int[3];
        int[] next = new int[3];
        for (int i = 0; i < n; i++) {
            int a = scanner.nextInt();
            int b = scanner.nextInt();
            int c = scanner.nextInt();
            next[0] = a + Math.max(prev[1], prev[2]);
            next[1] = b + Math.max(prev[0], prev[2]);
            next[2] = c + Math.max(prev[0], prev[1]);
            int[] tmp = next;
            next = prev;
            prev = tmp;
        }
        System.out.println(Math.max(prev[0], Math.max(prev[1], prev[2])));
    }
}

File: src/codeforces/contest1899/ProblemC.kt
package codeforces.contest1899

import java.io.BufferedReader
import java.io.InputStream
import java.io.InputStreamReader
import java.util.*
import kotlin.math.max

fun main(args: Array<String>) {
    val tests = nextInt()
    for (t in 0 until tests) {
        val n = nextInt()
        val arr = nextInts(n)
        println(solve(arr))
    }
}

fun solve(arr: IntArray): Long {
    var ans = Long.MIN_VALUE
    var cur = 0L
    var prev = 2
    for (num in arr) {
        if (num.and(1) == prev) cur = num.toLong()
        else cur += num
        ans = max(ans, cur)
        if (cur <= 0) {
            cur = 0
            prev = 2
        } else prev = num.and(1)
    }
    return ans
}

private class MyReader(inputStream: InputStream) {
    private val reader = BufferedReader(InputStreamReader(inputStream))
    private var tokenizer: StringTokenizer? = null

    fun next(): String {
        while (true) {
            tokenizer.let {
                if (it == null || !it.hasMoreTokens()) {
                    tokenizer = StringTokenizer(reader.readLine())
                } else {
                    return it.nextToken()
                }
            }
        }
    }
}

private val reader = MyReader(System.`in`)
private fun next() = reader.next()
private fun nextInt() = next().toInt()
private fun nextLong() = next().toLong()
private fun nextDouble() = next().toDouble()
private fun nextStrings(n: Int) = Array(n) { next() }
private fun nextInts(n: Int) = IntArray(n) { nextInt() }
private fun nextLongs(n: Int) = LongArray(n) { nextLong() }
private fun nextDoubles(n: Int) = DoubleArray(n) { nextDouble() }